tests/rule_based_profiler/test_profiler_run_anonymizer.py

```diff
@@ -0,0 +1,328 @@
+                        "expectation_type": "expect_column_pair_values_A_to_be_greater_than_B",
+                    }
+                ],
+                "anonymized_name": "5a83f3728393d6519a197cffdccd50ff",
+                "anonymized_parameter_builders": [
+                    {
+                        "anonymized_name": "9349ed253aba01f4ecf190af61018a11",
+                        "class_name": "MetricMultiBatchParameterBuilder",
+                    }
+                ],
+            },
+            {
+                "anonymized_domain_builder": {"class_name": "TableDomainBuilder"},
+                "anonymized_expectation_configuration_builders": [
+                    {
+                        "class_name": "DefaultExpectationConfigurationBuilder",
+                        "expectation_type": "expect_column_values_to_be_between",
+                    }
+                ],
+                "anonymized_name": "0bac2cecbb0cf8bb704e86710941434e",
+                "anonymized_parameter_builders": [
+                    {
+                        "anonymized_name": "b7719efec76c6ebe30230fc1ec023beb",
+                        "class_name": "MetricMultiBatchParameterBuilder",
+                    }
+                ],
+            },
+        ],
+        "config_version": 1.0,
+        "rule_count": 2,
+        "variable_count": 1,
+    }
+
+
+def test_anonymize_profiler_run_with_batch_requests_in_builder_attrs(
+    profiler_run_anonymizer: ProfilerRunAnonymizer,
+    usage_stats_profiler_config: dict,
+):
+    # Add batch requests to fixture before running method
+    batch_request: dict = {
+        "datasource_name": "my_datasource",
+        "data_connector_name": "my_basic_data_connector",
+        "data_asset_name": "my_data_asset",
+    }
+    rules: Dict[str, dict] = usage_stats_profiler_config["rules"]
+    rule: dict = rules["rule_1"]
+    rule["domain_builder"]["batch_request"] = batch_request
+    rule["parameter_builders"][0]["batch_request"] = batch_request
+
+    anonymized_result: dict = profiler_run_anonymizer.anonymize_profiler_run(
+        **usage_stats_profiler_config
+    )
+    assert anonymized_result == {
+        "anonymized_name": "5b6c98e19e21e77191fb071bb9e80070",
+        "config_version": 1.0,
+        "anonymized_rules": [
+            {
+                "anonymized_domain_builder": {
+                    "anonymized_batch_request": {
+                        "anonymized_batch_request_required_top_level_properties": {
+                            "anonymized_data_asset_name": "eac128c5824b698c22b441ada61022d4",
+                            "anonymized_data_connector_name": "123a3221fc4b65014d061cce4a71782e",
+                            "anonymized_datasource_name": "df78ebde1957385a02d8736cd2c9a6d9",
+                        },
+                        "batch_request_optional_top_level_keys": [
+                            "data_connector_query"
```

Comment:
Interesting that this `data_connector_query` is showing up even though you didn't provide it in the batch_request - do you have an idea of why that might be?

Reply:
I am not! That's a great point though.

If we look at the call stack, here's what goes on:
```python
# batch_request_anonymizer - get_batch_request_from_acceptable_arguments
{
    "datasource_name": "datasource_name",
    "data_connector_name": "my_basic_data_connector",
    "data_asset_name": "data_asset_name",
    "data_connector_query": {"batch_filter_parameters": {}},
    "limit": None,
    "batch_spec_passthrough": {}
}
```
```python
# batch_request_anonymizer - anonymize_batch_request_properties
{
    "datasource_name": ...,
    "data_connector_name": ...,
    "data_asset_name": ...,
    "data_connector_query": {},
}
```

Is this unexpected behavior?

Reply:
It is unexpected - when `data_connector_query` is not provided, I would expect it to not appear in the `batch_request_optional_top_level_keys`. Any idea why it is getting added?

Reply:
It isn't being removed by `deep_filter_properties_iterable` - I've added a workaround by essentially running that filter function as many times as needed to remove all falsy values but I think this may be a bug in our code.

Reply:
It looks like it doesn't get cleared out from `deep_filter_properties_iterable`.

We take in our `**kwargs` and create a batch request using `get_batch_request_from_acceptable_arguments`. This sets:
```python
    "data_connector_query": {"batch_filter_parameters": {}},
```

The filtering function results in:
```python
    "data_connector_query": {},
```

Since this key is part of the top-level (even though it has no associated value), it gets included. This seems to be a bug with the filtering function or the actual BatchRequestAnonymizer.